lib/vpc/vpc.ts

```diff
@@ -1,88 +1,76 @@
-import { Stack, StackProps, Tags } from 'aws-cdk-lib';
-import { SubnetType, Vpc } from 'aws-cdk-lib/aws-ec2';
-import { StringListParameter, StringParameter } from 'aws-cdk-lib/aws-ssm';
+import { aws_ec2, aws_ssm, Stack, StackProps, Tags, Fn, CfnOutput } from 'aws-cdk-lib';
 import { Construct } from 'constructs';
-import { VPCConfig } from '../../interfaces/lib/vpc/interfaces';
+import { VPCConfig } from '@harshadbhatia/improved-cdk-constructs/interfaces/lib/vpc/interfaces';
+
 
 export class VPCStack extends Stack {
-  vpc: Vpc;
+  vpc: aws_ec2.IVpc
   config: VPCConfig;
 
   constructor(scope: Construct, id: string, config: VPCConfig, props?: StackProps) {
     super(scope, id, props);
-
-    this.config = config;
-    this.createVPC();
-    this.addTags();
+    this.config = config
+    this.vpc = this.getVpc()
     this.createParams();
+    this.addTags();
   }
 
-  createVPC(): void {
-    this.vpc = new Vpc(this, 'VPC', {
-      maxAzs: 4,
-      cidr: '10.0.0.0/16',
-      natGateways: this.config.natGateways,
-      subnetConfiguration: [
-        {
-          name: 'Public',
-          subnetType: SubnetType.PUBLIC,
-        },
-        {
-          name: 'Private',
-          subnetType: SubnetType.PRIVATE_WITH_NAT,
-        },
-        {
-          name: 'Protected',
-          subnetType: SubnetType.PRIVATE_ISOLATED,
-        },
-        {
-          name: 'Spare',
-          subnetType: SubnetType.PRIVATE_ISOLATED,
-        },
+  getVpc(): aws_ec2.IVpc {
+    return aws_ec2.Vpc.fromVpcAttributes(this, 'VPC', {
+      vpcId: Fn.importValue('citadel-VPC').toString(),
+      availabilityZones: ['ap-southeast-2a','ap-southeast-2b','ap-southeast-2c'],
+      privateSubnetIds: [
```

Comment:
Hmm the VPC stack is meant to be generic. This is no longer generic, perhaps we can add additional props to allow import instead.

Reply:
Move vpc out of this repo.